candidate.py

```diff
@@ -135,6 +140,62 @@ def normalize_name(name):
     normalized += " %s" % suffix
   return normalized
 
+def normalize_district(district):
+  """Translate a district into an ordinal and state.
+  Args:
+    district: (str) a string like "Alabama 1" or "14"
+  returns:
+    (str, str): a state and ordinal district, e.g., ("Alabama", "1st")
+  """
+  if not district:
+    return (None, None)
+  state = None
+  match = re.search(r"^(\d+)$", district)
+  if match is not None:
+    # The district is just a number
+    number = match.group(1).lstrip('0') or "0"
+  else:
+    # Let's see if it's a state and a number
+    district_re = r"^(.*)\W+(\d+|at-large)$"
+    match = re.search(district_re, district)
+    number = ""
+    state = ""
+    if match is not None:
+      state = match.group(1)
+      number = match.group(2)
+    else:
+      print "Error: [%s] didn't match [%s]" % (district_re, district)
+      return (None, None)
+
+  if number == "at-large":
+    suffix = ""
+  elif re.match("^1.$", number):
+    suffix = "th"
+  elif number[-1] == "1":
+    suffix = "st"
+  elif number[-1] == "2":
+    suffix = "nd"
+  elif number[-1] == "3":
+    suffix = "rd"
+  else:
+    suffix = "th"
+  return (state, number + suffix)
+
+def normalize_state(state):
```

Comment:
maybe name the variable something that makes it more clear it takes a state abbreviation? state_abbr or postal_code or something? It would just make it slightly easier to read and understand the function statement quickly.